app/(playground)/p/[agentId]/beta-proto/artifact/server-actions.ts

```diff
@@ -71,38 +70,28 @@ ${sourcesToText(sources)}
 				prompt: params.userPrompt,
 				schema: artifactSchema,
 				onFinish: async (result) => {
-					const duration = performance.now() - startTime;
-					const measurementScope = await getCurrentMeasurementScope();
-					const isR06User = await isRoute06User();
-					generation.end({
-						output: result,
-					});
-
-					logger.info(
-						{
-							externalServiceName: "openai",
-							tokenConsumed: {
-								input: result.usage.promptTokens,
-								output: result.usage.completionTokens,
-							},
-							duration,
-							measurementScope,
-							isR06User,
+					await withTokenMeasurement(
+						logger,
+						async () => {
+							generation.end({ output: result });
+							await lf.shutdownAsync();
+							waitUntil(
+								new Promise((resolve) =>
```

Comment:
Yes, it'll be better if we can simplify these repetitive lines.
Although `waitUntil()` did not work when wrapped by `withMeasurement()`, but I'll consider other approach in another PR.
Let me merge this PR to avoid conflicting with another PR, thanks! @shige

Reply:
I think it's good. 👍 @Rindrics

Reply:
> I feel that the processing from L79-90 may lead to increased hardcoding in each file. Moving this part to something like lib/opentelemetry/wrapper.ts could improve the clarity, don't you think?

will be resolved by 551d82d8b15ec77f188f3645ebe6ca0374f27505